src/odemis/acq/test/fastem_test.py

```diff
@@ -109,6 +115,362 @@ def test_overview_acquisition(self):
+        y_fields = 4
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+        coordinates = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)  # in m
+        roc = fastem.FastEMROC("roc_name", coordinates)
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+        roa = fastem.FastEMROA(roa_name,
+                               coordinates,
+                               roc,
+                               self.asm,
+                               self.multibeam,
+                               self.descanner,
+                               self.mppc
+                               )
+        cell_res = self.mppc.cellCompleteResolution.value
+        dwell_time = self.multibeam.dwellTime.value
+        flyback = self.descanner.physicalFlybackTime  # extra time per line scan
+
+        # calculate expected field acquisition time
+        # (number of pixels per line * dwell time + flyback time) * number of lines
+        estimated_field_acq_time = (cell_res[0] * dwell_time + flyback) * cell_res[1]
+        # get field acquisition time
+        field_acq_time = roa.estimate_single_field_time()
+
+        self.assertAlmostEqual(estimated_field_acq_time, field_acq_time)
+
+    def test_estimate_roa_time(self):
+        """Check that the estimated time for one ROA (megafield) is calculated correctly."""
+        x_fields = 3
+        y_fields = 4
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+        coordinates = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)  # in m
+        roc = fastem.FastEMROC("roc_name", coordinates)
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+        roa = fastem.FastEMROA(roa_name,
+                               coordinates,
+                               roc,
+                               self.asm,
+                               self.multibeam,
+                               self.descanner,
+                               self.mppc
+                               )
+        cell_res = self.mppc.cellCompleteResolution.value
+        dwell_time = self.multibeam.dwellTime.value
+        flyback = self.descanner.physicalFlybackTime  # extra time per line scan
+
+        # FIXME take field indices into account, if ROA not an integer multiple ROA time incorrect!
+        # calculate expected roa (megafield) acquisition time
+        # (number of pixels per line * dwell time + flyback time) * number of lines * number of cells in x and y
+        estimated_roa_acq_time = (cell_res[0] * dwell_time + flyback) * cell_res[1] * x_fields * y_fields
+        # get roa acquisition time
+        roa_acq_time = roa.estimate_roa_time()
+
+        self.assertAlmostEqual(estimated_roa_acq_time, roa_acq_time)
+
+    def test_calculate_field_indices_int(self):
```

Comment:
I feel like this test and the next (`test_calculate_field_indices`) don't fit on this class, and should belong to the TestFastEMAcquisition class. Here I would expect a test that actually tests the correct field indices are returned. Something like:

```
x_fields = 3
y_fields = 2
expected_indices = [(0,0), (1,0), (2,0), (0,1), (1,1), (2,1)]  # I think this is correct, but correct me if I'm wrong.

field_indices = roa.calculate_field_indices()

self.assertEqual(expected_indices, field_indices)  # I always forget if this works for lists or not

```

Reply:
I agree.